browser-extension/mv2/src/client/js/ruleExecutionHandler.js

```diff
@@ -129,6 +128,10 @@ RQ.RuleExecutionHandler.notifyRuleAppliedToWidget = (ruleId) => {
   const testRuleWidget = document.querySelector("rq-test-rule-widget");
 
   if (testRuleWidget?.getAttribute("applied-status") === "false") {
+    if (testRuleWidget.getAttribute("rule-id") === ruleId) {
+      testRuleWidget.setAttribute("applied-status", true);
+    }
+
```

Comment:
Why are we doing this?

Reply:
Why are we doing this? This used to work before, what changed here?

Reply:
There exists a race condition in the current version. It solves that. Sometimes applied rules were not indicated